### PR TITLE
Add bazel test Throttle resources for CI

### PR DESCRIPTION
**Pull request by yxun**
Signed-off-by: Yuanlin <yuanlin.xu@redhat.com>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by yxun** [comment]
When I run maistra-2.2 maistra/envoy maistra/run-ci.sh in an OCP pod, I set the env variable CI="true" in the container.
At the beginning of the log , I can see --config=ci-config

```
++ '[' -n true ']'
++ COMMON_FLAGS+=' --config=ci-config '
++ '[' -n '' ']'
++ '[' -n '' ']'
```

However, in the middle of the run-ci.sh exection, my pod CPU jumped and reached request limits.Then OCP killed an integration test execution when it requested more CPU than the limit.

image

From the reference, it looks like we only set the resource throttle in the `bazel build ...` part. But we haven't set any resource throttle in the `bazel test ...` part yet.
https://bazel.build/reference/command-line-reference

**Comment by oschaaf** [comment]
We discussed this on Slack earlier: I feel that tuning these resource limits should be done only when our CI actually needs it. Having said that, maybe a bazel comment indicating that said config line only throttles build and not test execution is worth it?

I'm curious if the CI failure is a flake of some sorts, I have a hard time associating them to this change.. so:

/retest

File: maistra/bazelrc [hunk 1]
```diff
@@ -20,6 +20,9 @@ build:aarch64 --linkopt=-fuse-ld=lld
 build:ci-config --local_ram_resources=12288
 build:ci-config --local_cpu_resources=6
 build:ci-config --jobs=3
+test:ci-config --local_ram_resources=12288
```
Comment (jwendell) [inline maistra/bazelrc:23]: This should not be necessary as `test` inherits is configuration from `build`: https://bazel.build/run/bazelrc#option-defaults

**Comment by oschaaf** [comment]
Closing this: with earlier testing, me and @jwendell  have observed that the change proposed here actually is a no-op (as suggested in a comment above).

**Comment by yxun** [comment]
https://bazel.build/docs/user-manual#running-tests
"By default, this command performs simultaneous build and test activity"
That might explain why the cpu usage is over the config value 6.
I will try some workaround thanks.

**Pull request closed without merging**